open-craft/openedx-blockstore-relay
# LX-53 Upload all the bundle files at once, instead of one at a time.

**Description:** When transferring a Block to a Blockstore bundle, upload all the related files at once.

This allows Blockstore to create a single BundleVersion for the change.

**JIRA:** SE-281 

**Dependencies:**

* https://github.com/open-craft/blockstore/pull/8 (merged)

**Installation instructions:**

1. Install this branch in the CMS shell.
1. Run the Blockstore service so the CMS shell can access its REST API.
1. Ensure that the URL for the Blockstore is present in `settings.BLOCKSTORE_API_URL` (see [`openedx_blockstore_relay.settings`](https://github.com/open-craft/openedx-blockstore-relay/tree/master/openedx_blockstore_relay/settings)).

**Testing instructions:**

1. Create a Collection in Blockstore, and note its UUID:
    http://localhost:8888/api/v1/collections/
1. From the Studio shell, run the management command to upload a block to the given collection, e.g.:
   ```bash
   ./manage.py cms transfer_to_blockstore \
        --block-key="block-v1:edX+DemoX+Demo_Course+type@course+block@course" \
        --collection-uuid=a530f2a9-2b0c-4d59-8f41-01878a414102 \
        --settings=devstack_docker
   ```
1. Check that Blockstore has created only a single BundleVersion for this upload, e.g.,
   http://localhost:8888/api/v1/bundle_versions/?bundle__collection__uuid=a530f2a9-2b0c-4d59-8f41-01878a414102

**Reviewers:**

**Merge checklist:**
